stan/math/prim/arr/err/is_matching_size.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+#define STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+
+#include <stan/math/prim/scal/err/is_size_match.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Check if two structures are the same size.
```

Comment:
Let's update the documentation here. For our guidelines, see https://github.com/stan-dev/stan/wiki/How-to-Write-Doxygen-Doc-Comments

I think what makes sense here is to change the first line to something like:
"Returns <code>true</code> if two variables are the same size."